examples/camera/main.go

```diff
@@ -0,0 +1,262 @@
+// Copyright 2020 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build example jsgo
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"image"
+	_ "image/png"
+	"log"
+	"math"
+
+	"golang.org/x/image/math/f64"
+
+	"github.com/hajimehoshi/ebiten"
+	"github.com/hajimehoshi/ebiten/ebitenutil"
+	"github.com/hajimehoshi/ebiten/examples/resources/images"
+)
+
+const (
+	screenWidth  = 240
+	screenHeight = 240
+)
+
+const (
+	tileSize = 16
+	tileXNum = 25
+)
+
+const (
+	worldSizeX = screenWidth / tileSize
+	worldSizeY = screenHeight / tileSize
+)
+
+var (
+	tilesImage *ebiten.Image
+)
+
+func init() {
+	// Decode image from a byte slice instead of a file so that
+	// this example works in any working directory.
+	// If you want to use a file, there are some options:
+	// 1) Use os.Open and pass the file to the image decoder.
+	//    This is a very regular way, but doesn't work on browsers.
+	// 2) Use ebitenutil.OpenFile and pass the file to the image decoder.
+	//    This works even on browsers.
+	// 3) Use ebitenutil.NewImageFromFile to create an ebiten.Image directly from a file.
+	//    This also works on browsers.
+	img, _, err := image.Decode(bytes.NewReader(images.Tiles_png))
+	if err != nil {
+		log.Fatal(err)
+	}
+	tilesImage, _ = ebiten.NewImageFromImage(img, ebiten.FilterDefault)
+}
+
+type Camera struct {
+	ViewPort    f64.Vec2
+	Position    f64.Vec2
+	Zoom        f64.Vec2
+	Rotation    float64
+	worldMatrix ebiten.GeoM
+}
+
+func (c *Camera) String() string {
+	return fmt.Sprintf(
+		"T: %.1f, R: %.1f, S: %.1f",
+		c.Position, c.Rotation, c.Zoom,
+	)
+}
+
+func (c *Camera) viewportCenter() f64.Vec2 {
+	return f64.Vec2{
+		c.ViewPort[0] * 0.5,
+		c.ViewPort[1] * 0.5,
+	}
+}
+
+func (c *Camera) updateMatrix() {
```

Comment:
I'd create a `worldMatrix` function to return a GeoM, so that the camera would be a little stateless.